server/application/application.go

```diff
@@ -1826,7 +1826,33 @@ func (s *Server) ListLinks(ctx context.Context, req *application.ListAppLinksReq
 		return nil, fmt.Errorf("failed to read application deep links from configmap: %w", err)
 	}
 
-	finalList, errorList := deeplinks.EvaluateDeepLinksResponse(*obj, deepLinks)
+	var clstObj *unstructured.Unstructured
+	if err := argo.ValidateDestination(ctx, &a.Spec.Destination, s.db); err != nil {
```

Comment:
Should we also validate that the specified cluster is allowed by the app's project?

Reply:
Would it not be validated already as part of app creation?

Reply:
Only if app creation were done via the API. Someone could `kubectl apply` an invalid Application.

Reply:
I see, updated PR to have rbac check for cluster and project resource being accessed